### PR TITLE
feat(lock): better error messages on failure

### PR DESCRIPTION
**Pull request by mkenigs**
Instead of just saying that a group failed to resolve, say which package failed to resolve in which input.

Also factored out some shared data in test cases.

### CODE REVIEW

**Review by zmitchell — APPROVED** [review]

File: tests/environment.cc [hunk 1]
```diff
@@ -882,6 +870,43 @@ test_createLockfile_both()
...
+    }
+  catch ( flox::FloxException & err )
+    {
+      EXPECT_EQ(
+        err.what(),
+        std::string(
+          "resolution failure: failed to resolve some package(s):\n"
+          "  in default group:\n"
+          "    failed to resolve `not-a-package' in input "
+          "`github:NixOS/nixpkgs/e8039594435c68eb4f780f3e9bf3972a7399c4b1'" ) );
```
Comment (mkenigs) [inline tests/environment.cc:904]: @zmitchell do we need to tweak the newlines or indentation so this plays nicely with our error formatting in Rust, or does it already look okay as is?

Comment (zmitchell) [reply tests/environment.cc:904]: Good catch. It's kind of hard to tell whether a newline in the message here will inherit the indentation that the Rust side applies to each error in the chain of errors. Can you try setting `PKGDB_BIN` in flox at runtime to see how it looks?

Comment (mkenigs) [reply tests/environment.cc:904]: It seems context message is not being printed at all:
```
> PKGDB=./bin/pkgdb SHELL=bash flox --debug activate
...
[DEBUG] [posix] + locking manifest with command: "/Users/matthew/pkgdb/main/bin/pkgdb" "manifest" "lock" "--ga-registry" "/Users/matthew/debug/.flox/env/manifest.toml"
[ERROR] [flox] locking manifest failed: resolution failure
> "/Users/matthew/pkgdb/main/bin/pkgdb" "manifest" "lock" "--ga-registry" "/Users/matthew/debug/.flox/env/manifest.toml"
resolution failure: failed to resolve some package(s):
  in default group:
    failed to resolve `bad1' in input `github:NixOS/nixpkgs/f741f8a839912e272d7e87ccf4b9dbc6012cdaf9'
```

Comment (mkenigs) [reply tests/environment.cc:904]: Is this the issue? https://github.com/flox/flox/blob/93785d22b78e59e7a5dcaa4cdf1e18956c608e50/crates/flox-rust-sdk/src/models/pkgdb_errors.rs#L68

Comment (zmitchell) [reply tests/environment.cc:904]: I don't think that's the issue. If a type implementing `std::error::Error` implements the `source` method then `anyhow` will continue going down that chain of sources until it gets a `None` back. This line is just printing the message for this "link" in the chain.

**Pull request merged**